### DISCUSSION TITLE
How to generate package.json with all the dependencies

### DISCUSSION THREAD

**Discussion by Mat-moran** [nrwl/nx#18810] [General]
Hi 😄  ,

I was diging  around to setup deployment with fly.io in a remix app and I arrived to this related issue https://github.com/nrwl/nx-labs/issues/31 . 

I was trying to follow the steps to create an executor that generate the `package.json` to be able to install all the dependencies in the Dockerfile when I realised that there in not `createPackageJson()` function anymore with `@nx/devkit`

How can I achieve the same functionality? Or how can I build my Dockerfile if the packages are not in the build context of the image? Should I copy all the monorepo inside the Dockerfile and do the instalation with the `nx cli` inside the Dockerfile? and have the Dockerfile in the top level directory? 

I need a bit of light please 🙏 😄

**Comment by JoshMentzer** [comment]
I found the createPackageJson export in @nx/js - not sure if it was an intended move, as the recipe in the documentation still mentions it coming from @nx/devkit.  Hope that helps!
